client/mycart/src/app/customer/feedback/feedback.component.html

```diff
@@ -1 +1,45 @@
\ No newline at end of file
+<h1 class="text-center text-primary my-4">Feedback</h1>
+<div class="row">
+    <div class="col-md-6 offset-md-3">
+        <form #form="ngForm" (ngSubmit)="onSubmit(form)">
+            <div class="m-3">
+                <label for="email" class="mb-2">Email</label>
+                <input type="email" id="email" class="form-control text-center text-dark mb-2" name="email"
+                    placeholder="Enter Email" pattern="^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$" minlength="5"
+                    maxlength="70" #email="ngModel" [(ngModel)]="model.email" required>
+                <div class="text-danger" *ngIf="email.invalid && email.touched">
+                    <div *ngIf="email.errors?.['minlength']">Email is too short.</div>
+                    <div *ngIf="email.errors?.['required']">Email is required.</div>
+                    <div *ngIf="email.errors?.['maxlength']">Email is too long.</div>
+                    <div *ngIf="email.errors?.['pattern']">Email is invalid.</div>
+                </div>
+            </div>
+            <div class="m-3">
+                <label for="fullname" class="mb-2">Full Name</label>
+                <input type="fullname" id="password" class="form-control text-center text-dark mb-2" name="fullname"
+                    placeholder="Enter full Name" minlength="6" maxlength="25" [(ngModel)]="model.fullname"
+                    #fullname="ngModel" required>
+                <div class="text-danger" *ngIf="fullname.invalid && fullname.touched">
+                    <div *ngIf="fullname.errors?.['minlength']">The given name is too small.</div>
+                    <div *ngIf="fullname.errors?.['maxlength']">The given name is too long.</div>
+                    <div *ngIf="fullname.errors?.['required']">Full name is required.</div>
+                </div>
+            </div>
+            <div class="m-3">
+                <label for="feedback" class="mb-2">Feedback</label>
+                <textarea type="feedback" id="feedback" class="form-control text-center text-dark mb-2" name="feedback"
+                    placeholder="Give feedback here" minlength="4" maxlength="25" [(ngModel)]="model.feedback"
+                    #feedback="ngModel" required></textarea>
+                <div class="text-danger" *ngIf="feedback.invalid && feedback.touched">
+                    <div *ngIf="feedback.errors?.['minlength']">The given feedbackis too small.</div>
+                    <div *ngIf="feedback.errors?.['maxlength']">The given geedback is too long.</div>
+                    <div *ngIf="feedback.errors?.['required']">Feedback is required.</div>
+                </div>
+            </div>
+            <div class="d-flex justify-content-center align-items-center mt-5 mb-5">
+                <button type="submit" class="btn btn-primary"
+                    [disabled]="email.invalid && fullname.invalid && feedback.invalid">Submit</button>
```

Comment:
Use form model's invalid property to disable the button.

Reply:
Done.